### PR TITLE
WIP: watcherEx

### PR DESCRIPTION
**Pull request by Shivansh-yadav13**
- discussed with @nodece 
- I have created this to get reviews from everyone

### CODE REVIEW

File: src/watcherEx.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
+import { RedisClusterConnection, RedisConnection, RedisClient } from './redis';
```
Comment (hsluoyz) [inline src/watcherEx.ts:1]: apache header

File: src/watcherEx.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import { ClusterNode, ClusterOptions, RedisOptions } from 'ioredis';
+import { Model } from 'casbin';
+
+export interface WatcherExOptions extends RedisOptions {
```
Comment (hsluoyz) [inline src/watcherEx.ts:5]: test?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@Shivansh-yadav13 finish this: https://github.com/node-casbin/redis-watcher/issues/28 first

File: src/watcherEx.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+    const client = await watcherEx.subConnection.getRedisClient();
+    // @ts-ignore - subscribe is exists.
+    await client.subscribe(watcher.channel);
+    client.on('message', (chan: string, msg: any) => {
+      if (chan !== watcherEx.channel) {
+        return;
+      }
+
+      if (msg.type === 'updateForAddPolicy' && watcherEx.callbackForAddPolicy) watcherEx.callbackForAddPolicy();
```
Comment (nodece) [inline src/watcherEx.ts:149]: Could you make these types to a const or enum to avoid hard-code?

**Review by nodece — CHANGES_REQUESTED** [review]
There is nothing wrong with this PR overall, could you add a test for watcherEx?

**Comment by hsluoyz** [comment]
@Shivansh-yadav13 fix errors:

image

**Comment by hsluoyz** [comment]
@Shivansh-yadav13 plz fix commits, they are messed up

image

**Comment by Shivansh-yadav13** [comment]
I'm sorry rebasing accidentally caused this, I'll make fresh PR

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Not reusable

**Comment by Shivansh-yadav13** [comment]
I'm sorry rebasing accidentally caused this, I'll make fresh PR

**Pull request merged**